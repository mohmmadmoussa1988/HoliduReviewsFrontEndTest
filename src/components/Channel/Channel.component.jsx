import React from 'react';
import { ChannelWrapper } from './Channel.styles';
import AIRBNB from '../../assets/AIRBNB.svg';
import HOLIDU from '../../assets/HOLIDU.svg';
import BOOKINGCOM from '../../assets/BOOKINGCOM.svg';
const Channel = ({ children }) => {
    const Image = (children) => {
        switch (children) {
            case 'AIRBNB': return AIRBNB;
            case 'HOLIDU': return HOLIDU;
            case 'BOOKINGCOM': return BOOKINGCOM;
            default:
                return '';
        }
    }
    return (
        <>
            <ChannelWrapper><img src={Image(children)} alt={children} /></ChannelWrapper>
        </>
    )
}

export default Channel;