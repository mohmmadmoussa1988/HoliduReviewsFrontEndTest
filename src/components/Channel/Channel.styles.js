import styled from "styled-components";
export const ChannelWrapper = styled.div`
  padding-top: 5px;
  img {
    height: 16px;
    margin-top: 4px;
  }
`;
