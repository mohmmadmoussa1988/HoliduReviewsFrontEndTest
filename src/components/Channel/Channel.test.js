import React from "react";
import { configure, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Channel from "./Channel.component";

describe(`<Channel />`, () => {
  it("should render Channel image", () => {
    const wrapper = shallow(<Channel />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
