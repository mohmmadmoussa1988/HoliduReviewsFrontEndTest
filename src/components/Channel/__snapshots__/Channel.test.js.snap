// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Channel /> should render Channel image 1`] = `
<Fragment>
  <styled.div>
    <img
      src=""
    />
  </styled.div>
</Fragment>
`;
